Test `x86_64-gnu` in PR CI, not x86_64-gnu-llvm-14

jyn514 opened the pull request:
This is a pre-requisite for https://github.com/rust-lang/rust/pull/112143, which wants to start using download-rustc in PRs. download-rustc doesn't allow providing an external LLVM.

r? @ghost

jyn514 committed:
Test `x86_64-gnu` in PR CI, not x86_64-gnu-llvm-14

This is a pre-requisite for https://github.com/rust-lang/rust/pull/112143, which wants to start
using download-rustc in PRs. download-rustc doesn't allow providing an external LLVM.

rust-log-analyzer commented:
The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5172522731/jobs/9316948344) [(plain)](https://github.com/rust-lang/rust/commit/f1cb26cfef5232e80353caf55fec844898d8e464/checks/13995331097/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Successfully tagged rust-ci:latest
Built container sha256:2c103b03e5ad1c56bcc95516531028cb2c8c95b09bc79586c15f999b632d1c2e
Uploading finished image to https://ci-caches.rust-lang.org/docker/0e428aacea371664b80befb7fc24bf4332422d150349ecc29750fa67a734738ee582fd6bea96b8e41b6f4a0a90e2a950d9520cc81be4657f555124bd108cee11

<botocore.awsrequest.AWSRequest object at 0x7f683a6d3950>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=x86_64-gnu]
[CI_JOB_NAME=x86_64-gnu]
---
   Compiling unwind v0.0.0 (/checkout/library/unwind)
   Compiling profiler_builtins v0.0.0 (/checkout/library/profiler_builtins)
The following warnings were emitted during compilation:

warning: In file included from /usr/lib/gcc/x86_64-linux-gnu/9/include/limits.h:194,
warning:                  from /usr/lib/gcc/x86_64-linux-gnu/9/include/syslimits.h:7,
warning:                  from /usr/lib/gcc/x86_64-linux-gnu/9/include/limits.h:34,
warning:                  from /checkout/src/llvm-project/compiler-rt/lib/builtins/int_lib.h:93,
warning:                  from /checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c:13:
warning: /usr/include/limits.h:26:10: fatal error: bits/libc-header-start.h: No such file or directory
warning:    26 | #include <bits/libc-header-start.h>
warning: compilation terminated.

error: failed to run custom build command for `compiler_builtins v0.1.92`


Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/release/build/compiler_builtins-59afb72d651a3b75/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=build.rs
  cargo:compiler-rt=/cargo/registry/src/index.crates.io-6f17d22bba15001f/compiler_builtins-0.1.92/compiler-rt
  cargo:rustc-cfg=feature="unstable"
  cargo:rustc-cfg=feature="mem-unaligned"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c
  cargo:rustc-cfg=__absvdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvsi2.c
  cargo:rustc-cfg=__absvsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvti2.c
  cargo:rustc-cfg=__absvti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvdi3.c
  cargo:rustc-cfg=__addvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvsi3.c
  cargo:rustc-cfg=__addvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvti3.c
  cargo:rustc-cfg=__addvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/ashldi3.S
  cargo:rustc-cfg=__ashldi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/ashrdi3.S
  cargo:rustc-cfg=__ashrdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/clzdi2.c
  cargo:rustc-cfg=__clzdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/clzsi2.c
  cargo:rustc-cfg=__clzsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/clzti2.c
  cargo:rustc-cfg=__clzti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/cmpdi2.c
  cargo:rustc-cfg=__cmpdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/cmpti2.c
  cargo:rustc-cfg=__cmpti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ctzdi2.c
  cargo:rustc-cfg=__ctzdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ctzsi2.c
  cargo:rustc-cfg=__ctzsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ctzti2.c
  cargo:rustc-cfg=__ctzti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/divdc3.c
  cargo:rustc-cfg=__divdc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/divdi3.S
  cargo:rustc-cfg=__divdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/divsc3.c
  cargo:rustc-cfg=__divsc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/divxc3.c
  cargo:rustc-cfg=__divxc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/extendhfsf2.c
  cargo:rustc-cfg=__extendhfsf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ffsti2.c
  cargo:rustc-cfg=__ffsti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/floatdixf.S
  cargo:rustc-cfg=__floatdixf="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/floatundixf.S
  cargo:rustc-cfg=__floatundixf="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/int_util.c
  cargo:rustc-cfg=__int_util="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/lshrdi3.S
  cargo:rustc-cfg=__lshrdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/moddi3.S
  cargo:rustc-cfg=__moddi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/muldc3.c
  cargo:rustc-cfg=__muldc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/muldi3.S
  cargo:rustc-cfg=__muldi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulsc3.c
  cargo:rustc-cfg=__mulsc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvdi3.c
  cargo:rustc-cfg=__mulvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvsi3.c
  cargo:rustc-cfg=__mulvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvti3.c
  cargo:rustc-cfg=__mulvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulxc3.c
  cargo:rustc-cfg=__mulxc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negdf2.c
  cargo:rustc-cfg=__negdf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negdi2.c
  cargo:rustc-cfg=__negdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negsf2.c
  cargo:rustc-cfg=__negsf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negti2.c
  cargo:rustc-cfg=__negti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvdi2.c
  cargo:rustc-cfg=__negvdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvsi2.c
  cargo:rustc-cfg=__negvsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvti2.c
  cargo:rustc-cfg=__negvti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/paritydi2.c
  cargo:rustc-cfg=__paritydi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/paritysi2.c
  cargo:rustc-cfg=__paritysi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/parityti2.c
  cargo:rustc-cfg=__parityti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountdi2.c
  cargo:rustc-cfg=__popcountdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountsi2.c
  cargo:rustc-cfg=__popcountsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountti2.c
  cargo:rustc-cfg=__popcountti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/powixf2.c
  cargo:rustc-cfg=__powixf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvdi3.c
  cargo:rustc-cfg=__subvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvsi3.c
  cargo:rustc-cfg=__subvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvti3.c
  cargo:rustc-cfg=__subvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/truncdfhf2.c
  cargo:rustc-cfg=__truncdfhf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/truncsfhf2.c
  cargo:rustc-cfg=__truncsfhf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ucmpdi2.c
  cargo:rustc-cfg=__ucmpdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ucmpti2.c
  cargo:rustc-cfg=__ucmpti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/udivdi3.S
  cargo:rustc-cfg=__udivdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/i386/umoddi3.S
  cargo:rustc-cfg=__umoddi3="optimized-c"
  TARGET = Some("i686-unknown-linux-gnu")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CC_i686-unknown-linux-gnu
  CC_i686-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  CC_i686_unknown_linux_gnu = Some("sccache x86_64-linux-gnu-gcc")
  CFLAGS_i686-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_i686_unknown_linux_gnu
  CFLAGS_i686_unknown_linux_gnu = Some("-ffunction-sections -fdata-sections -fPIC -m32 -march=i686")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CC_i686-unknown-linux-gnu
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  CC_i686_unknown_linux_gnu = Some("sccache x86_64-linux-gnu-gcc")
  CFLAGS_i686-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_i686_unknown_linux_gnu
  CFLAGS_i686_unknown_linux_gnu = Some("-ffunction-sections -fdata-sections -fPIC -m32 -march=i686")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CC_i686-unknown-linux-gnu
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  cargo:rerun-if-env-changed=CC_i686_unknown_linux_gnu
  CC_i686_unknown_linux_gnu = Some("sccache x86_64-linux-gnu-gcc")
  CFLAGS_i686-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_i686_unknown_linux_gnu
  CFLAGS_i686_unknown_linux_gnu = Some("-ffunction-sections -fdata-sections -fPIC -m32 -march=i686")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "sccache" "x86_64-linux-gnu-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m32" "-march=i686" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m32" "-march=i686" "-fno-builtin" "-fvisibility=hidden" "-ffreestanding" "-DVISIBILITY_HIDDEN" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/i686-unknown-linux-gnu/release/build/compiler_builtins-077d2f81b7b48254/out/45c91108d938afe8-absvdi2.o" "-c" "/checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c"
  cargo:warning=In file included from /usr/lib/gcc/x86_64-linux-gnu/9/include/limits.h:194,
  cargo:warning=                 from /usr/lib/gcc/x86_64-linux-gnu/9/include/syslimits.h:7,
  cargo:warning=                 from /usr/lib/gcc/x86_64-linux-gnu/9/include/limits.h:34,
  cargo:warning=                 from /checkout/src/llvm-project/compiler-rt/lib/builtins/int_lib.h:93,
  cargo:warning=                 from /checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c:13:
  cargo:warning=/usr/include/limits.h:26:10: fatal error: bits/libc-header-start.h: No such file or directory
  cargo:warning=   26 | #include <bits/libc-header-start.h>
  cargo:warning=      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=compilation terminated.

  --- stderr



  error occurred: Command "sccache" "x86_64-linux-gnu-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m32" "-march=i686" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m32" "-march=i686" "-fno-builtin" "-fvisibility=hidden" "-ffreestanding" "-DVISIBILITY_HIDDEN" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/i686-unknown-linux-gnu/release/build/compiler_builtins-077d2f81b7b48254/out/45c91108d938afe8-absvdi2.o" "-c" "/checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c" with args "x86_64-linux-gnu-gcc" did not execute successfully (status code exit status: 1).

warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:00:30
```

</details>

rust-log-analyzer commented:
The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5172815925/jobs/9317513386) [(plain)](https://github.com/rust-lang/rust/commit/3b136d42f173f7412f3004239dd7f8a01def3c2b/checks/13995995373/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Successfully tagged rust-ci:latest
Built container sha256:ef6af664bfc32d0f4ac9ccab88dc948057373cd35a46f18d1d8d20acc22b57f3
Uploading finished image to https://ci-caches.rust-lang.org/docker/9879777d97c297fdd43ab363069d73ab1e628b1b5f77a425fdda3d4533ea2623cd43c673dbad7c4357045a07bce1c10a13eb242ad49529edf8d2f1d6ba7690f9

<botocore.awsrequest.AWSRequest object at 0x7f66f446fe90>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=x86_64-gnu]
[CI_JOB_NAME=x86_64-gnu]
---

 finished in 5.950 seconds
Build completed successfully in 0:01:15
+ ../x.ps1 --stage 2 test tests/ui --pass=check --host= --target=i686-unknown-linux-gnu
/usr/bin/env: 'pwsh': No such file or directory
```

</details>

jyn514 committed:
don't try to cross-compile sanitizer tests

this was a pre-existing latent bug, we just didn't have any CI builders exercising it.

fixes the following errors:
```
 ---- [ui] tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt1 stdout ----

error in revision `opt1`: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--cfg" "opt1" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/auxiliary" "-Zsanitizer=address" "-Clto=thin" "-Copt-level=1"
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

failures:
    [ui] tests/ui/sanitize/badfree.rs
    [ui] tests/ui/sanitize/address.rs
    [ui] tests/ui/sanitize/use-after-scope.rs
    [ui] tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt0
    [ui] tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt1
```

rust-log-analyzer commented:
The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5173032463/jobs/9317921973) [(plain)](https://github.com/rust-lang/rust/commit/76ab44f0424332de8a3153ba7677a1bc27b04ce2/checks/13996488976/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Successfully tagged rust-ci:latest
Built container sha256:c1f017776818d0393cf782e915809a76b47de577140e9200ae92a3e59617d4e3
Uploading finished image to https://ci-caches.rust-lang.org/docker/9b7a975ad0a0997f5c6c76c8f322a94e9c5aae2078123f7be08c89c66991214340bc0620f92d68dc2d1da5a3cdc19dd74b34199c9aee7649e11c3a12eb66ebfa

<botocore.awsrequest.AWSRequest object at 0x7fcf317e7090>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=x86_64-gnu]
[CI_JOB_NAME=x86_64-gnu]
---
...................................................................iiiiiii...i....i.i... 11880/15092
........................................................................................ 11968/15092
..............................................iiiiiiiiii.iii............................ 12056/15092
i....................................................................................... 12144/15092
.................iiii..i.i.iiiiiii..iiii..iiiii.......F....F.......F.................... 12232/15092
........................................................................................ 12408/15092
.........................................F.............................................. 12496/15092
........................................................................................ 12584/15092
......................................................................................i. 12672/15092
---
---- [ui] tests/ui/sanitize/badfree.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/badfree.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/badfree/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/badfree/auxiliary" "-Z" "sanitizer=address" "-O"
stdout: none
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   |
   = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/bin:/opt/microsoft/powershell/7:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "x86_64-linux-gnu-gcc" "-m32" "/tmp/rustcqmGd2X/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a" "-Wl,--no-whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/badfree/a.badfree.16967218b800787b-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/badfree/auxiliary" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,-Bdynamic" "-lstd-319e89dc37a430ed" "-Wl,-Bstatic" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/libcompiler_builtins-71cded8efb2a6273.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/badfree/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,--enable-new-dtags" "-Wl,-z,origin"
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

error: aborting due to previous error
------------------------------------------



---- [ui] tests/ui/sanitize/address.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/address.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a" "-A" "unused" "-Crpath" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/auxiliary" "-Z" "sanitizer=address" "-O" "-g"
stdout: none
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   |
   = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/bin:/opt/microsoft/powershell/7:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "x86_64-linux-gnu-gcc" "-m32" "/tmp/rustccIQPPO/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a" "-Wl,--no-whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a.address.3f41a11184bdbc52-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/auxiliary" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,-Bdynamic" "-lstd-319e89dc37a430ed" "-Wl,-Bstatic" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/libcompiler_builtins-71cded8efb2a6273.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,--enable-new-dtags" "-Wl,-z,origin"
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

error: aborting due to previous error
------------------------------------------



---- [ui] tests/ui/sanitize/use-after-scope.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/use-after-scope.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/use-after-scope/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/use-after-scope/auxiliary" "-Zsanitizer=address"
stdout: none
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   |
   = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/bin:/opt/microsoft/powershell/7:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "x86_64-linux-gnu-gcc" "-m32" "/tmp/rustcCMFrup/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a" "-Wl,--no-whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/use-after-scope/a.use_after_scope.c09c8d74cb44b59c-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/use-after-scope/auxiliary" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,-Bdynamic" "-lstd-319e89dc37a430ed" "-Wl,-Bstatic" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/libcompiler_builtins-71cded8efb2a6273.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/use-after-scope/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,--enable-new-dtags" "-Wl,-z,origin"
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

error: aborting due to previous error
------------------------------------------



---- [ui] tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt0 stdout ----

error in revision `opt0`: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--cfg" "opt0" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/auxiliary" "-Zsanitizer=address" "-Clto=thin" "-Copt-level=0"
stdout: none
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   |
   = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/bin:/opt/microsoft/powershell/7:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "x86_64-linux-gnu-gcc" "-m32" "/tmp/rustcA7ga8K/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a" "-Wl,--no-whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.3q9auv0abkrzn5xq.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.addr2line-0cb0728354932f81.addr2line.f4b8e81554815aa2-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.adler-d296d4f72dfc2ef8.adler.55dcc74b4acda8ae-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.alloc-dc8558520c41e8a3.alloc.50763989e2851a43-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.cfg_if-a2c6241b077619f7.cfg_if.653e6f4f2c178df2-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.core-f8fe8721a8368db9.core.560235037c5aa337-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.gimli-617738e595a821de.gimli.b00ab6b4c4cc45f3-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.hashbrown-e9f77429dac046d8.hashbrown.f1c0c72b9626d886-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.libc-8832f2e2bf28c29e.libc.bbf596db12671875-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.memchr-e3a800a8111b42f9.memchr.5d88cd2006497ff6-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.miniz_oxide-93128b9194c667b8.miniz_oxide.2e70e79926a09cac-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.new_llvm_pass_manager_thin_lto.ee8c321b0fad48d7-cgu.0.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.object-4f7b7c623dc3e7e4.object.5c664a3e9678fc1e-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.panic_unwind-4a34851c0d1ab2b6.panic_unwind.1b8f7ae906bcd138-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.rustc_demangle-e752601e2b3a4f4d.rustc_demangle.b969a52c9711cbf8-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.rustc_std_workspace_alloc-63e894f537abfe05.rustc_std_workspace_alloc.5118e21c6583eeec-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.rustc_std_workspace_core-3fa62bd80c43e0d3.rustc_std_workspace_core.c8e49cfa41bd769f-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.std-319e89dc37a430ed.std.dc2ccfd7c89db86a-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.std_detect-272c20bba7a986fd.std_detect.407efe74ff2711d9-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a.unwind-863209c4993b3d04.unwind.bd43175f6899aeae-cgu.0.rcgu.o.rcgu.o" "-Wl,--as-needed" "-L" "/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/auxiliary" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/libcompiler_builtins-71cded8efb2a6273.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,--enable-new-dtags" "-Wl,-z,origin"
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

error: aborting due to previous error
------------------------------------------



---- [ui] tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt1 stdout ----

error in revision `opt1`: test compilation failed although it shouldn't!
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--cfg" "opt1" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/auxiliary" "-Zsanitizer=address" "-Clto=thin" "-Copt-level=1"
stdout: none
--- stderr -------------------------------
error: linking with `x86_64-linux-gnu-gcc` failed: exit status: 1
   |
   = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/bin:/opt/microsoft/powershell/7:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "x86_64-linux-gnu-gcc" "-m32" "/tmp/rustcmdo2Nx/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a" "-Wl,--no-whole-archive" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.3q9auv0abkrzn5xq.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.addr2line-0cb0728354932f81.addr2line.f4b8e81554815aa2-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.adler-d296d4f72dfc2ef8.adler.55dcc74b4acda8ae-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.alloc-dc8558520c41e8a3.alloc.50763989e2851a43-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.cfg_if-a2c6241b077619f7.cfg_if.653e6f4f2c178df2-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.core-f8fe8721a8368db9.core.560235037c5aa337-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.gimli-617738e595a821de.gimli.b00ab6b4c4cc45f3-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.hashbrown-e9f77429dac046d8.hashbrown.f1c0c72b9626d886-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.libc-8832f2e2bf28c29e.libc.bbf596db12671875-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.memchr-e3a800a8111b42f9.memchr.5d88cd2006497ff6-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.miniz_oxide-93128b9194c667b8.miniz_oxide.2e70e79926a09cac-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.new_llvm_pass_manager_thin_lto.ee8c321b0fad48d7-cgu.0.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.object-4f7b7c623dc3e7e4.object.5c664a3e9678fc1e-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.panic_unwind-4a34851c0d1ab2b6.panic_unwind.1b8f7ae906bcd138-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.rustc_demangle-e752601e2b3a4f4d.rustc_demangle.b969a52c9711cbf8-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.rustc_std_workspace_alloc-63e894f537abfe05.rustc_std_workspace_alloc.5118e21c6583eeec-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.rustc_std_workspace_core-3fa62bd80c43e0d3.rustc_std_workspace_core.c8e49cfa41bd769f-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.std-319e89dc37a430ed.std.dc2ccfd7c89db86a-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.std_detect-272c20bba7a986fd.std_detect.407efe74ff2711d9-cgu.0.rcgu.o.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a.unwind-863209c4993b3d04.unwind.bd43175f6899aeae-cgu.0.rcgu.o.rcgu.o" "-Wl,--as-needed" "-L" "/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/auxiliary" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/libcompiler_builtins-71cded8efb2a6273.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt1/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/i686-unknown-linux-gnu/lib" "-Wl,--enable-new-dtags" "-Wl,-z,origin"
   = note: x86_64-linux-gnu-gcc: error: /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i686-unknown-linux-gnu/lib/librustc-nightly_rt.asan.a: No such file or directory

error: aborting due to previous error
------------------------------------------

```

</details>

rust-log-analyzer commented:
The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5173244242/jobs/9318335839) [(plain)](https://github.com/rust-lang/rust/commit/63cb54d6c5a67b4bcfb2408b8fff09cbb101a94f/checks/13996993492/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Successfully tagged rust-ci:latest
Built container sha256:f6d274ccd30582a28d7310cb0f9d1afaae980bdf633a6dd5819ab9512cbf6a84
Uploading finished image to https://ci-caches.rust-lang.org/docker/9b7a975ad0a0997f5c6c76c8f322a94e9c5aae2078123f7be08c89c66991214340bc0620f92d68dc2d1da5a3cdc19dd74b34199c9aee7649e11c3a12eb66ebfa

<botocore.awsrequest.AWSRequest object at 0x7fc369f5fa90>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=x86_64-gnu]
[CI_JOB_NAME=x86_64-gnu]
---
failures:

---- [ui] tests/ui/sanitize/address.rs stdout ----

error: error pattern ' 'xs' (line 13) <== Memory access at offset' not found!
command: cd "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address" && RUST_TEST_THREADS="16" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a"
stdout: none
--- stderr -------------------------------
=================================================================
=================================================================
==97472==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffcc6a59eb0 at pc 0x556f9e820cfc bp 0x7ffcc6a59e50 sp 0x7ffcc6a59e48
READ of size 4 at 0x7ffcc6a59eb0 thread T0
    #0 0x556f9e820cfb  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfecfb) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)
    #1 0x556f9e820ab2  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfeab2) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)
    #2 0x556f9e820ad8  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfead8) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)
    #3 0x7f0d86cc5779  (/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-4b2c915dc5c16928.so+0x9b779) (BuildId: 09b86d35842644ae33d2d8d521a264814128acee)
    #4 0x556f9e820df4  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfedf4) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)
    #5 0x7f0d868c2082  (/lib/x86_64-linux-gnu/libc.so.6+0x24082) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)
    #6 0x556f9e72a28d  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0x828d) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)

Address 0x7ffcc6a59eb0 is located in stack of thread T0 at offset 80 in frame
    #0 0x556f9e820b3f  (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfeb3f) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf)

  This frame has 2 object(s):
    [32, 40) ''
    [64, 80) 'xs' (line 14) <== Memory access at offset 80 overflows this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow (/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitize/address/a+0xfecfb) (BuildId: 63f5f072e34e698392ac850c04fc75d3686865bf) 
Shadow bytes around the buggy address:
  0x7ffcc6a59c00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a59c80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a59d00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a59d80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a59e00: 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1 f1 f1
=>0x7ffcc6a59e80: f8 f2 f2 f2 00 00[f3]f3 00 00 00 00 00 00 00 00
  0x7ffcc6a59f00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a59f80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a5a000: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a5a080: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x7ffcc6a5a100: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==97472==ABORTING



failures:
```

</details>

Nilstrieb commented:
When I changed codegen tests, CI running old LLVM was useful because it immediately made me aware that I forgot to make them old LLVM compatible (typed pointers 😿). But that problem should get a lot smaller once we finally drop support for typed pointer LLVM.
But I also clearly see the benefit for using download-rustc, so it's probably worth it. cc @nikic anyways if you have opinions

nikic commented:
@Nilstrieb I think that's okay, because the issue goes both ways. If we're testing llvm-14, we don't see codegen test failures for the current LLVM version in PRs, so one of the configurations will not be covered either way, and I'm not sure the llvm-14 configuration is inherently more valuable from that perspective. Fixing llvm-N failures is usually just a matter of slapping `min-llvm-version` on the test, so doesn't require much trial and error.

Nilstrieb commented:
Well, I assume that `src/llvm-project` LLVM will be tested locally, so both configurations are tested. But yeah, not a big concern.
